src/precice/Tooling.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <iosfwd>
+#include <string>
+
+namespace precice {
+
+/** @brief Contains the preCICE tooling API
+ *
+ * The contained methods allow to query internal information of preCICE
+ * without having to create a \ref precice::SolverInterface.
+ *
+ * @note These functions are not exposed via the bindings
+ *
+ * @see \ref precice::getVersionInformation which is exposed via the bindings
+ */
+namespace tooling {
+
+/** @brief Prints the configuration reference as Markdown.
+ *
+ * @param[inout] out the stream to write the result to
+ */
+void printConfigAsMD(std::ostream &out);
```

Comment:
```suggestion
void printConfigReferenceAsMD(std::ostream &out);
```
It does not print a valid config. Thinking one step ahead we might also print actual configs in the future.

Reply:
True. The longer function name makes a single function now more appealing to me.

Reply:
Merged in 29fb82c